app/(blobs)/(setup)/_components/UploadThingTokenForm.tsx

```diff
@@ -21,7 +22,7 @@ export const UploadThingTokenForm = () => {
   }: {
     uploadThingToken: string;
   }) => {
-    await submitUploadThingForm({ uploadThingToken });
+    await action(uploadThingToken);
```

Comment:
I think part of why we're not getting good UX is that we aren't using the built in form state management hooks for server actions. A good future improvement would be to create the `<Form>` component that we discussed, which would handle this functionality automatically.